recipes/create_pwfile.rb

```diff
@@ -0,0 +1,95 @@
+#
+# Cookbook Name:: dovecot
+# Recipe:: create_pwfile
+# Author:: Xabier de Zuazo (<xabier@zuazo.org>)
+# Copyright:: Copyright (c) 2013-2014 Onddo Labs, SL.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+extend Chef::Mixin::ShellOut
+
+include_recipe 'dovecot::from_package'
+
+# The file credentials should be like:
+# user:password:uid:gid:(gecos):home:(shell):extra_fields
+# We ignore gecos and shell, all the others are included in the script but
+# uid,gid,home,extra_Fields can be nil
+# user:pass:<num>:<num>::<string>::<Extras string>
+
+credentials = []
+credentials_updated = false
+
+local_creds = {}
+if ::File.exist?(node['dovecot']['conf']['password_file'])
+  passwordfile = File.open(
+    node['dovecot']['conf']['password_file'], File::RDONLY | File::CREAT, 640
+  )
+
+  passwordfile.readlines.each do |line|
+    (user, crypt, uid, gid, gecos, homedir, shell, extra_fields) \
+      = line.strip.split(':')
+    local_creds[user] = if line.strip.split(':').length == 2
+                          [crypt, uid, gid, gecos, homedir, shell, extra_fields]
+                        else
+                          [crypt]
+                        end
```

Comment:
refactored :) now inside libs
this accepts any type of userdb entry
there have been 2 ways:
user:pass (old)
current:
user:pass:uid:gid:gecos:homedir:shell:extra < some of these doesn't work
this is to force importing properly both, but now after I made some changes it should work better, with less complexity